### PR TITLE
added toggle access command

### PR DESCRIPTION
**Pull request by g--o**
#268
@JRoy Something like this?

### CODE REVIEW

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * @author g--o
+ *
+ * Command for toggling chest access
+ */
+public class AccessToggle implements CommandExecutor {
+    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
+        if (cmd.getName().equals("csaccess")) {
+            String resultMsg = "Access to shops is ";
```
Comment (JRoy) [inline src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: The messages from this command will need to follow the same format as other commands (Look at `/cstoggle` for reference)

File: src/main/java/com/Acrobot/ChestShop/Permission.java [hunk 2]
```diff
@@ -3,11 +3,13 @@
...
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
+import org.bukkit.permissions.PermissionAttachment;
+import org.bukkit.permissions.PermissionAttachmentInfo;
 
 import java.util.Optional;
 
 /**
- * @author Acrobot
+ * @author Acrobot, g--o
```
Comment (JRoy) [inline src/main/java/com/Acrobot/ChestShop/Permission.java]: Don't update the author tag here

File: src/main/resources/plugin.yml [hunk 3]
```diff
@@ -25,6 +25,9 @@ commands:
   cstoggle:
     description: Toggle messages to the owner of a shop
     usage: /<command>
+  csaccess:
+    description: Toggle access to chest
```
Comment (JRoy) [inline src/main/resources/plugin.yml]: A better description would be "Toggle moderator access to chestshops"

File: src/main/java/com/Acrobot/ChestShop/ChestShop.java [hunk 4]
```diff
@@ -1,10 +1,7 @@
 package com.Acrobot.ChestShop;
 
 import com.Acrobot.Breeze.Configuration.Configuration;
-import com.Acrobot.ChestShop.Commands.Give;
-import com.Acrobot.ChestShop.Commands.ItemInfo;
-import com.Acrobot.ChestShop.Commands.Toggle;
-import com.Acrobot.ChestShop.Commands.Version;
+import com.Acrobot.ChestShop.Commands.*;
```
Comment (Phoenix616) [inline src/main/java/com/Acrobot/ChestShop/ChestShop.java]: I know that there are already some places that use wildcard imports but I would prefer it if we could avoid introducing new ones, especially if it increases the diff.

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java (hunk 1 above)
Comment (Phoenix616) [reply src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: I agree, messages should be all configurable via the message config file.

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            Player player = (Player) sender;
+            String permNode = Permission.OTHER_NAME_ACCESS + ".*";
+
+            if (PlayerInteract.canOpenOtherShops(player)) { // @FIX: why is this deprecated?
```
Comment (Phoenix616) [inline src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: > why is this deprecated?

See the comment on the method: You should be using the actual utility method mentioned there directly. It was deprecated as all usage of that method was removed from the plugin in b853afbd5223d9f6885601c60e5ec438e6721926.

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+                Permission.setPermission(player, permNode, false);
+
+                resultMsg += ChatColor.DARK_RED + "disabled";
+            } else {
+                Permission.setPermission(player, permNode, true);
```
Comment (Phoenix616) [inline src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: Please find a better way of handling the toggling besides modifying the actual permission attachments of the player as that could lead to issues with certain permission plugins and recalculating permissions tends to be slow in general. I suggest an approach similar to the notification toggling with `/cstoggle`.

File: src/main/java/com/Acrobot/ChestShop/ChestShop.java (hunk 4 above)
Comment (g--o) [reply src/main/java/com/Acrobot/ChestShop/ChestShop.java]: if you work with IDEA, ctrl+alt+o optimizes the imports and might save you build time

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java (hunk 5 above)
Comment (g--o) [reply src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: well but now i use it lmao, it has no logical replacement

File: src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java (hunk 6 above)
Comment (g--o) [reply src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: I was asked in the issue to toggle this particular permission, it's up to @JRoy I guess.
I don't think making a list is better, considering the amount of people using this is pretty minimal.
If anything, the problem lies on the fact that your plugin doesn't use PermissionProvider interface of any sort, and does not provide a setPermission which I tried to resolve here.
What's the use of permissions if you practically encode permissions via blacklist and whitelist in the end? Pretty silly

Comment (g--o) [reply src/main/java/com/Acrobot/ChestShop/Commands/AccessToggle.java]: Not saying this is neccessarily a permenant solutionm but given the amount of users that have access to the command it'll do for now

**Comment by JRoy** [comment]
@g--o if you're not gonna fix the issues Phoenix requested, at least add me as a contributor to your fork and I'll do it.

**Comment by g--o** [comment]
> @g--o if you're not gonna fix the issues Phoenix requested, at least add me as a contributor to your fork and I'll do it.

np, as long as you agree I've provided what you wished for

**Comment by JRoy** [comment]
@g--o You've provided what I've asked and the bounty will still be awarded to you :P

**Comment by JRoy** [comment]
@Phoenix616 your changes have been applied, let me know if there's anything else that needs fixing

**Comment by Phoenix616** [comment]
Fixed some more small things myself and merged it. Thanks!

**Pull request merged**